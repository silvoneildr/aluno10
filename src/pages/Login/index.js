import React, { useState } from 'react';

import Card from 'Components/Card';
import Form from './styles';

export default function Login() {
  const [email, setEmail] = useState({ value: '', hasError: false, touched: false });
  const [password, setPassword] = useState({ value: '', hasError: false, touched: false });

  const handleEmail = ({ target: { value: emailValue } }) => {
    if (!emailValue && email.touched) {
      setEmail({
        ...email,
        value: emailValue,
        hasError: true
      });
    } else {
      setEmail({
        ...email,
        value: emailValue,
        hasError: false,
        touched: true
      });
    }
  };

  const handlePassword = ({ target: { value: passwordValue } }) => {
    if (!passwordValue && password.touched) {
      setPassword({
        ...password,
        value: passwordValue,
        hasError: true
      });
    } else {
      setPassword({
        ...password,
        value: passwordValue,
        hasError: false,
        touched: true
      });
    }
  };

  return (
    <Form>
      <Card>
        <input
          placeholder="E-mail"
          type="email"
          name="email"
          onChange={handleEmail}
          id="email"
          value={email.value}
        />
        {email.hasError && <p>E-mail obrigatório</p>}
        <input
          placeholder="Senha"
          type="password"
          name="password"
          onChange={handlePassword}
          id="password"
          value={password.value}
          data-testid="password"
        />
        {password.hasError && <p>Senha obrigatória</p>}
        <button type="submit">Entrar</button>
      </Card>
    </Form>
  );
}
