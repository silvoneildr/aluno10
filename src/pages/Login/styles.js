import styled from 'styled-components';

const Form = styled.form`
  height: 100%;
  width: 100%;
`;

export default Form;
