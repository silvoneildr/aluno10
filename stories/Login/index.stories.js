import React from 'react';

import { storiesOf } from '@storybook/react';

import Login from 'Pages/Login';
import { Wrapper } from '../index.stories';

storiesOf('Pages|Login', module).add('Screen', () => (
  <Wrapper>
    <Login />
  </Wrapper>
));
