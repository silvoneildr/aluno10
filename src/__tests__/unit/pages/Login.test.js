import React from 'react';
import Login from 'Pages/Login';
import { render } from 'react-testing-library';

describe('Login', () => {
  it('should render correctly', () => {
    const { container } = render(<Login />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
