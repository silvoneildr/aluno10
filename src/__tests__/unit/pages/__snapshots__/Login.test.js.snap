// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render correctly 1`] = `
.c1 {
  width: 100%;
  height: 100%;
  padding: 20px;
  border-radius: 10px;
}

.c0 {
  height: 100%;
  width: 100%;
}

<form
  class="c0"
>
  <div
    class="c1"
  >
    <input
      id="email"
      name="email"
      placeholder="E-mail"
      type="email"
      value=""
    />
    <input
      data-testid="password"
      id="password"
      name="password"
      placeholder="Senha"
      type="password"
      value=""
    />
    <button
      type="submit"
    >
      Entrar
    </button>
  </div>
</form>
`;
